### PR TITLE
Remove support for signaling test result by return values

### PR DESCRIPTION
**Pull request by xpconanfan**
Also add an entry in .gitignore.

### CODE REVIEW

**Comment by xpconanfan** [comment]
It does beg the question: what should happen if a test returns something?

File: mobly/controllers/android_device.py [hunk 1]
```diff
@@ -366,7 +366,7 @@ def __init__(self, serial="", host_port=None, device_port=8080):
         self.adb_logcat_file_path = None
         self.adb = adb.AdbProxy(serial)
         self.fastboot = fastboot.FastbootProxy(serial)
-        if not self.is_bootloader:
+        if not self.is_bootloader and self.is_rootable:
```
Comment (adorokhine) [inline mobly/controllers/android_device.py]: Seems like this PR also mixes in the adb root change?
I'm guessing you didn't start it on a clean branch?

Comment (xpconanfan) [inline mobly/controllers/android_device.py]: ohhh, each branch needs to be started on master, is that so?.... ughhh

**Comment by adorokhine** [comment]
Still vestiges of this left in other methods, like setup_class:
"To signal setup failure, return False or raise an exception."

And the return path...

File: mobly/base_test.py [hunk 2]
```diff
@@ -126,15 +126,15 @@ def _setup_class(self):
...
+        self.setup_class()
 
     def setup_class(self):
         """Setup function that will be called before executing any test case in
         the test class.
 
-        To signal setup failure, return False or raise an exception. If
-        exceptions were raised, the stack trace would appear in log, but the
-        exceptions would not propagate to upper levels.
+        To signal setup failure, user asserts or raise your own exception. The
```
Comment (adorokhine) [inline mobly/base_test.py]: use asserts

File: mobly/base_test.py [hunk 3]
```diff
@@ -126,15 +126,15 @@ def _setup_class(self):
...
+        stack trace would appear in log, but the exceptions would not propagate
```
Comment (adorokhine) [inline mobly/base_test.py]: not sure what 'exceptions would not propagate' means... that makes it sound like setup_class can't fail the test.

File: mobly/base_test.py [hunk 4]
```diff
@@ -157,15 +157,15 @@ def _setup_test(self, test_name):
...
+        self.setup_test()
 
     def setup_test(self):
         """Setup function that will be called every time before executing each
         test case in the test class.
 
-        To signal setup failure, return False or raise an exception. If
-        exceptions were raised, the stack trace would appear in log, but the
-        exceptions would not propagate to upper levels.
+        To signal setup failure, user asserts or raise your own exception. The
```
Comment (adorokhine) [inline mobly/base_test.py]: use asserts

File: mobly/base_test.py [hunk 5]
```diff
@@ -157,15 +157,15 @@ def _setup_test(self, test_name):
...
+        stack trace would appear in log, but the exceptions would not propagate
```
Comment (adorokhine) [inline mobly/base_test.py]: same as above

File: mobly/base_test.py [hunk 6]
```diff
@@ -545,8 +533,7 @@ def run(self, test_names=None):
         class_record.test_begin()
         # Setup for the class.
         try:
-            if self._setup_class() is False:
-                asserts.fail("Failed to setup %s." % self.TAG)
+            self._setup_class()
         except Exception as e:
             self.log.exception("Failed to setup %s.", self.TAG)
```
Comment (adorokhine) [inline mobly/base_test.py]: _exec_func swallows the exception and returns False, but that return value is never used anywhere.

- Is that return still needed?
- Can we give _exec_func a better name to more clearly show that it swallows exceptions?

File: tests/mobly_base_class_test.py [hunk 7]
```diff
@@ -199,24 +199,6 @@ def test_something(self):
                             "Skipped 0, Unknown 0")
```
Comment (adorokhine) [inline tests/mobly_base_class_test.py]: We don't have any tests that tested "return False" from tests themselves? Only setup?

File: mobly/base_test.py (hunk 4 above)
Comment (xpconanfan) [inline mobly/base_test.py]: Which part do you want to replace?...

Comment (xpconanfan) [inline mobly/base_test.py]: oh, got it

File: tests/mobly_base_class_test.py (hunk 7 above)
Comment (xpconanfan) [inline tests/mobly_base_class_test.py]: I think I hated that behavior so much that I didn't write a test for it...

File: mobly/base_test.py (hunk 3 above)
Comment (xpconanfan) [inline mobly/base_test.py]: Ah that's a relic.
I think back in the days tests didn't fail when setup failed and that bug was there for a long time............. Removing it from doc strings

**Pull request merged**